matminer/featurizers/composition.py

```diff
@@ -2004,3 +2025,211 @@ def get_ideal_radius_ratio(self, n_neighbors):
+
+            self.writer = SummaryWriter(
+                # log_dir=(f"runs/{model_name}-r{run_id}_" "{date:%d-%m-%Y_%H-%M-%S}").format(
+                #     date=datetime.datetime.now()
+                # )
+                log_dir=f"runs/{model_name}-r{run_id}"
+            )
+        else:
+            self.writer = None
+
+        self.fitted = False
+
+        # multiprocessing doesn't play nicely with pytorch/CUDA instances
+        self.set_n_jobs(1)
+
+    def fit(self, X, y):
+
+        train_set = CompositionData(X, targets=y, task=self.task, emb=self.elem_emb)
+
+        train_generator = DataLoader(train_set, **self.data_params)
+
+        # if val_set is not None:
+        #     data_params.update({"batch_size": 16 * data_params["batch_size"]})
+        #     val_generator = DataLoader(self.val_set, **data_params)
+        # else:
+        #     val_generator = None
+
+        if self.model.task == "regression":
+            sample_target = torch.Tensor(train_set.targets)
+            self.normalizer.fit(sample_target)
+
+        # if (self.val_set is not None) and (self.model.best_val_score is None):
+        #     print("Getting Validation Baseline")
+        #     with torch.no_grad():
+        #         _, v_metrics = self.model.evaluate(
+        #             generator=self.val_generator,
+        #             criterion=self.criterion,
+        #             optimizer=None,
+        #             normalizer=self.normalizer,
+        #             action="val",
+        #             verbose=True,
+        #         )
+        #         if self.model.task == "regression":
+        #             val_score = v_metrics["MAE"]
+        #             print(f"Validation Baseline: MAE {val_score:.3f}\n")
+        #         elif self.model.task == "classification":
+        #             val_score = v_metrics["Acc"]
+        #             print(f"Validation Baseline: Acc {val_score:.3f}\n")
+        #         self.model.best_val_score = val_score
+
+        self.model.fit(
+            train_generator=train_generator,
+            val_generator=None,
+            # val_generator=val_generator,
+            optimizer=self.optimizer,
+            scheduler=self.scheduler,
+            epochs=self.epochs,
+            criterion=self.criterion,
+            normalizer=self.normalizer,
+            model_name=self.model_name,
+            run_id=self.run_id,
+            writer=self.writer,
+        )
+
+        self.fitted = True
+
+
+    def featurize(self, *comp):
+
+        assert self.fitted, "Please fit this featuriser before use"
+
+        test_set = CompositionData(comp, task=self.task)
+
+        test_generator = DataLoader(test_set, **self.data_params)
+        
+        # NOTE featurisation is faster on the cpu unless data is batched
+        # currently featurise_many doesn't use batches and instead applies
+        # the featuriser to each row in turn.
+        self.model.device = "cpu"
+        self.model.to("cpu")
```

Comment:
it's about 5x faster on CPU for me if we are running individual datapoints through the network